src/main/java/org/candlepin/async/tasks/CertificateCleanupJob.java

```diff
@@ -0,0 +1,132 @@
+        this.caCertCurator = Objects.requireNonNull(contentAccessCerts);
+        this.serialCurator = Objects.requireNonNull(serialCurator);
+    }
+
+    @Override
+    @Transactional
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        log.debug("Starting certificate cleanup.");
+        // Skipping entitlement certificates as those are handled by ExpiredPoolsCleanupJob
+        cleanupExpiredIdentityCerts();
+        cleanupExpiredContentAccessCerts();
+        cleanupCertificateSerials();
+        log.debug("Certificate cleanup successfully completed.");
+    }
+
+    private void cleanupExpiredIdentityCerts() {
+        List<ExpiredCertificate> allExpiredIdCertificates = this.identCertCurator.listAllExpired();
+        log.info("Cleaning up {} expired identity certificates.", allExpiredIdCertificates.size());
+        for (List<ExpiredCertificate> expiredCertificates : partition(allExpiredIdCertificates)) {
+            List<String> expiredCertIds = certIdsOf(expiredCertificates);
+            List<Long> expiredSerials = serialsOf(expiredCertificates);
+            List<String> consumersToUnlink = this.consumerCurator.findUuidsByIdentCerts(expiredCertIds);
+            int unlinkedConsumers = this.consumerCurator.unlinkIdCertificatesOf(consumersToUnlink);
+            log.debug("Unlinking identity certificates of {} consumers.", unlinkedConsumers);
+            this.identCertCurator.deleteByIds(expiredCertIds);
+            this.serialCurator.revokeByIds(expiredSerials);
+        }
+    }
+
+    private void cleanupExpiredContentAccessCerts() {
+        List<ExpiredCertificate> allExpiredCaCertificates = this.caCertCurator.listAllExpired();
+        log.info("Cleaning up {} expired content access certificates.", allExpiredCaCertificates.size());
+        for (List<ExpiredCertificate> expiredCertificates : partition(allExpiredCaCertificates)) {
+            List<String> expiredCertIds = certIdsOf(expiredCertificates);
+            List<Long> expiredSerials = serialsOf(expiredCertificates);
+            List<String> consumersToUnlink = this.consumerCurator
+                .findUuidsByContentAccessCerts(expiredCertIds);
+            int unlinkedConsumers = this.consumerCurator.unlinkCaCertificatesOf(consumersToUnlink);
+            log.debug("Unlinking content access certificates of {} consumers.", unlinkedConsumers);
+            this.caCertCurator.deleteByIds(expiredCertIds);
+            this.serialCurator.revokeByIds(expiredSerials);
+        }
+    }
+
+    private void cleanupCertificateSerials() {
+        int deleted = this.serialCurator.deleteRevokedExpiredSerials();
+        log.debug("Cleaning up {} expired and revoked certificate serials.", deleted);
+    }
+
+    private Iterable<List<ExpiredCertificate>> partition(List<ExpiredCertificate> list) {
+        return Iterables.partition(list, this.config.getInt(ConfigProperties.CERT_CLEANUP_BULK_SIZE));
```

Comment:
No need to do this with a custom config entry, especially with a default value of 1000 which is very low and might lead to too many queries. We already have an abstraction for the IN clause use case: `AbstractHibernateCurator.getInBlockSize()` which will use whatever the allowed limit for IN blocks is for the given underlying rdbms

Reply:
Done.